src/plugins/discover/server/ui_settings.ts

```diff
@@ -124,6 +125,19 @@ export const getUiSettings: (docLinks: DocLinksServiceSetup) => Record<string, U
     category: ['discover'],
     schema: schema.boolean(),
   },
+  [SHOW_LEGACY_FIELD_TOP_VALUES]: {
+    name: i18n.translate('discover.advancedSettings.showLegacyFieldStatsTitle', {
+      defaultMessage: "Show legacy Top Values for a field in the sidebar's popover",
+    }),
+    value: false,
+    type: 'boolean',
+    description: i18n.translate('discover.advancedSettings.showLegacyFieldStatsText', {
+      defaultMessage:
+        'This setting will calculate Top Values for a field based only on loaded records on Discover page. To use the new and more accurate view, turn off this option.',
+    }),
+    category: ['discover'],
```

Comment:
@gchaps This is for a new Advanced Setting to allow users to switch back to the old view if necessary.

Reply:
Is the new way the default?  If so, we should state that first and then contrast it with the old way. It needs some tweaking, but I am thinking of something like this:

> For best results, calculate top values using a sample of 5,000 records.  When disabled, base top values on the first 500 records that Discover loads.  

@ninoslavmiskovic

Reply:
@gchaps @kertal Right now the logic is different and the setting is called `discover:showLegacyFieldTopValues`. It's disabled by default. So users would need to enable it instead to switch back to the old view. Would you suggest to flip the logic for such flag and update its name and description?

Reply:
Took a look at other settings in the Discover area, and it seems we have a sort of pattern to let the user enable legacy behavior. One could argument if it's not better to let the user disable new stuff to get back to the old stuff. 
![Advanced_Settings_-_Elastic](https://user-images.githubusercontent.com/463851/189347043-4098734b-5c2a-4b42-b99e-f2280f891840.png)
I'd say to stick to that pattern, so to not change the current behavior

Reply:
@kertal Alright, thanks!
@gchaps Could you please update the copy?

Reply:
@kertal, @jughosta 

Do any of these work?

> To calculate the top values using up to 5,000 records, turn off this option.  By default, top values are based on the first 500 records that Discover loads.

>To calculate the top values using 5,000 records instead of 500, turn off this option.  A larger sample produces more accurate results.

>To calculate the top values using a sample of 5,000 records instead of the 500 records that Discover loads, turn off this option.  A larger sample produces more accurate results.

Reply:
@gchaps you need to turn on the legacy behavior (like we handle this with other Advanced settings entries) 

`To calculate the top values using 500 instead of 5,000 records, turn on this option.`

Don't think we need to mention that 5,000 is more accurate.